src/frontend/planner_test/tests/testdata/batch_index_join.yaml

```diff
@@ -58,3 +58,18 @@
     └─BatchLookupJoin { type: Inner, predicate: t.a = idx.c, output: all }
       └─BatchExchange { order: [], dist: UpstreamHashShard(t.a) }
         └─BatchScan { table: t, columns: [t.a, t.b], distribution: SomeShard }
+- name: test index join distribution derive
+  sql: |
+    set rw_batch_enable_lookup_join = true;
+    set query_mode = distributed;
```

Comment:
I guess the `batch_plan` must be distributed?

Reply:
Currently, our query execution mode by default is `local`, so we need to set it to `distributed`.

Reply:
Seems we always generate distributed plan. For local plan, we use the `local_batch_plan` field.
https://github.com/risingwavelabs/risingwave/blob/672c049254904aff99038366a57e3663aa0052b2/src/frontend/planner_test/src/lib.rs#L468-L479

Reply:
You are right. I can remove it in this https://github.com/risingwavelabs/risingwave/pull/6818 PR together.
```
set rw_batch_enable_lookup_join = true;
set query_mode = distributed;
```